uxarray/helpers.py

```diff
@@ -82,10 +84,149 @@ def determine_file_type(filepath):
         else:
             msg = str(e) + ': {}'.format(filepath)
     finally:
-        if msg != "":  # we did not catch this above
+        if msg != "":
             msg = "Unable to determine file type, mesh file not supported" + ': {}'.format(
                 filepath)
-            print(msg)
-            os._exit(0)
+            raise ValueError(msg)
 
     return mesh_filetype
+
+
+def spherical_to_cartesian_unit(lat, lon, r=6371):
+    lat, lon = np.deg2rad(lat), np.deg2rad(lon)
+    x = r * np.cos(lat) * np.cos(lon)  # x coordinate
+    y = r * np.cos(lat) * np.sin(lon)  # y coordinate
+    z = r * np.sin(lat)  # z coordinate
+
+    coord = np.array([x, y, z])
+    # make it coord on a sphere with unit radius
+    unit_coord = coord / np.linalg.norm(coord)
+
+    return unit_coord
+
+
+def spherical_to_cartesian_unit(node, r=6371):
+    """Converts spherical (lat/lon) coordinates to cartesian (x,y,z).
+
+    Final output is cartesian coordinates on a sphere of unit radius
+
+    Parameters:
+    -----------
+
+    node: a list consisting of lat and lon
+
+    Returns: numpy array
+        Cartesian coordinates of length 3
+    """
+    lon = node[0]
+    lat = node[1]
+    lat, lon = np.deg2rad(lat), np.deg2rad(lon)
+    x = r * np.cos(lat) * np.cos(lon)  # x coordinate
+    y = r * np.cos(lat) * np.sin(lon)  # y coordinate
+    z = r * np.sin(lat)  # z coordinate
+
+    coord = np.array([x, y, z])
+    # make it coord on a sphere with unit radius
+    unit_coord = coord / np.linalg.norm(coord)
+
+    return unit_coord
+
+
+# Calculate the area of all faces.
+def calculate_face_area(x, y, z, type="spherical"):
```

Comment:
Consider using numba for this and the jacobian function for a giant speedup.

Alternatively (and more fun!), could explore vectorizing using `tensordot`, `einsum` and friends from numpy (https://numpy.org/doc/stable/reference/routines.linalg.html#matrix-and-vector-products)